### PR TITLE
Misc improvements/v1

### PR DESCRIPTION
**Pull request by victorjulien**
Misc fixes and improvments.

Main: https://redmine.openinfosecfoundation.org/issues/6615

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16960

**Review by inashivb — COMMENTED** [review]
No opinion on https://github.com/OISF/suricata/pull/10016/commits/a3aea61554258e354e51a7c256c43a675eb7347f
Not sure about content inspection and bytejump analysis.
One nit ques inline.

File: rust/src/jsonbuilder.rs [hunk 1]
```diff
@@ -597,6 +597,27 @@ impl JsonBuilder {
...
+                return Err(JsonError::InvalidState);
+            }
+        }
+        self.push('"')?;
+        self.push_str(key)?;
+        self.push_str("\":")?;
+        self.push_str(&val.to_string())?;
+        Ok(self)
+    }
+
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply rust/src/jsonbuilder.rs:620]: Don't know. I just copied `set_uint`.

File: rust/src/jsonbuilder.rs [hunk 2]
```diff
@@ -597,6 +597,27 @@ impl JsonBuilder {
         Ok(self)
     }
 
+    /// Set a key and an signed integer type on an object.
```
Comment (jufajardini) [inline rust/src/jsonbuilder.rs:600]: nit: a signed integer

**Review by jufajardini — COMMENTED** [review]
Not confident to approve 2d51cfcba61f2, but looks good to me.
One nit on a function comment.
Overall, looks good to me.

**Review by jasonish — CHANGES_REQUESTED** [review]
The typo.

Also, anymore context for bytejump flag other than the "future" :)

**Comment by victorjulien** [comment]
> The typo.
> 
> Also, anymore context for bytejump flag other than the "future" :)

Right now using the wrong flag doesn't cause issues, as the flag's value is not used by bytejump flags. So there is no conflict. If in the future bytejump would add a new flag with the same value, it would cause unknown issues.

File: rust/src/jsonbuilder.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (inashivb) [reply rust/src/jsonbuilder.rs:620]: ah. I see. Thanks!

**Comment by victorjulien** [comment]
Comments addressed in #10042

**Pull request closed without merging**